Extension/src/LanguageServer/settings.ts

```diff
@@ -239,7 +239,23 @@ export class CppSettings extends Settings {
     public get defaultCustomConfigurationVariables(): { [key: string]: string } | undefined { return super.Section.get<{ [key: string]: string }>("default.customConfigurationVariables"); }
     public get useBacktickCommandSubstitution(): boolean | undefined { return super.Section.get<boolean>("debugger.useBacktickCommandSubstitution"); }
     public get codeFolding(): boolean { return super.Section.get<string>("codeFolding") === "Enabled"; }
-    public get legacyCompilerArgsBehavior(): boolean | undefined  { return super.Section.get<boolean>("legacyCompilerArgsBehavior"); }
+    private forceLegacyCompilerArgs: boolean | undefined;
+    public get legacyCompilerArgsBehavior(): boolean | undefined  {
+        if (this.forceLegacyCompilerArgs === undefined) {
+            this.forceLegacyCompilerArgs = false;
+            if (os.platform() === "darwin") {
+                const releaseParts: string[] = os.release().split(".");
+                if (releaseParts.length >= 1) {
+                    // wordexp doesn't work for older Mac OS's.
```

Comment:
Nit: put in comment explicit Mac version.

Reply:
I didn't think a comment was necessary because the code below is `< 16`.

Reply:
For devs who aren't familiar with Mac versions, adding version in comment makes things clear without needing to figure out where 16 is being compared to. Also, since comment mentions "older" I was expecting version number.

Don't mean to be nit picker, but it's nice to have clear code :).